task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,274 @@
+        SCAN_PROP=""
+
+        ca_bundle=/mnt/trusted-ca/ca-bundle.crt
+        if [ -f "$ca_bundle" ]; then
+          echo "INFO: Using mounted CA bundle: $ca_bundle"
+          cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+          update-ca-trust
+        fi
+
+        # Clone the source code from upstream repo
+        clone_unicontrol()
+        {
+            GIT_URL=$(echo "${FIND_UNICODE_CONTROL_GIT_URL}" | awk -F'#' '{print $1}')
+            REV=$(echo "${FIND_UNICODE_CONTROL_GIT_URL}" | awk -F'#' '{print $2}')
+
+            # Clone find-unicode-control repository
+            git clone --depth=1 "${GIT_URL}" >&2
+            status=$?
+            if [ "${status}" -ne 0 ]; then
+                echo "Error: Failed to clone the repository: ${GIT_URL}" >&2
+                return 1
+            else
+                echo "Message: Succeed to clone the repository: ${GIT_URL}" >&2
+            fi
+
+            if [[ -n "${REV}" ]]; then
+                # Get git url suffix
+                git_url_suffix=$(git -C ./find-unicode-control/ rev-parse HEAD)
+                if [[ "${git_url_suffix}" != "${REV}" ]]; then
+                    git -C ./find-unicode-control/ checkout --quiet "${REV}"
+                fi
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}"
+            else
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}#${git_url_suffix}"
+            fi
+
+            # Check usage of find-unicode-control to confirm the tool can work
+            ./find-unicode-control/find_unicode_control.py --help >&2
+            status=$?
+            if [ "${status}" -ne 0 ]; then
+                echo "Error: Failed to check usage of find-unicode-control" >&2
+                return 1
+            else
+                echo "Message: Succeed to check usage of find-unicode-control" >&2
+            fi
+        }
+
+        # Find unicode control
+        unicontrol_scan()
+        {
+            # Split the arguments into an array
+            args=("${FIND_UNICODE_CONTROL_ARGS}")
+
+            LANG=en_US.utf8 ./find-unicode-control/find_unicode_control.py "${args[@]}" "${SOURCE_CODE_DIR}/source" \
+                >raw_sast_unicode_check_out.txt \
+                2>raw_sast_unicode_check_out.log \
+                || FUC_EXIT_CODE=$?
+
+            if [[ -z "${FUC_EXIT_CODE}" || "${FUC_EXIT_CODE}" -eq 1 ]]; then
+                # No non-printable unicode characters found, or successfully found
+                echo "Message: Succeed to run find-unicode-control command" >&2
+            else
+                # Failed to run the command
+                echo "Error: failed to run find unicode control command" >&2
+                return 1
+            fi
+
+            # Translate the output format
+            sed -i raw_sast_unicode_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|'
+            status=$?
```

Comment:
See my previous comment.  The `status=$?` will not work with `set -e` because non-zero exit code of the previous command will terminate the shell.

Reply:
Updated